include/stomp_moveit/cost_functions.hpp

```diff
@@ -96,5 +91,76 @@ CostFn get_collision_cost_function(const std::shared_ptr<const planning_scene::P
 
   return cost_fn;
 }
+}  // namespace
+namespace costs
+{
+// Interpolation step size for collision checking (joint space, L2 norm)
+constexpr double COL_CHECK_DISTANCE = 0.05;
+constexpr double CONSTRAINT_CHECK_DISTANCE = 0.1;
+
+CostFn get_collision_cost_function(const std::shared_ptr<const planning_scene::PlanningScene>& planning_scene,
+                                   const moveit::core::JointModelGroup* group, double collision_penalty)
+{
+  const auto& joints = group ? group->getActiveJointModels() : planning_scene->getRobotModel()->getActiveJointModels();
+  const auto& group_name = group ? group->getName() : "";
+
+  StateValidatorFn collision_validator_fn = [=](const Eigen::VectorXd& positions) {
+    static moveit::core::RobotState state(planning_scene->getCurrentState());
+
+    // Update robot state values
+    set_joint_positions(positions, joints, state);
+    state.update();
+
+    return !planning_scene->isStateColliding(state, group_name);
+  };
+
+  return get_cost_function_from_state_validator(collision_validator_fn, COL_CHECK_DISTANCE, collision_penalty);
+}
+
+CostFn get_constraints_cost_function(const std::shared_ptr<const planning_scene::PlanningScene>& planning_scene,
```

Comment:
I am not very happy with this name because it is rather generic. Can we name it something like `get_constraints_msg_cost_function`?

Reply:
isn't that a bit redundant? msg is already part of the function signature